### PR TITLE
Feature/65 UI deploy

### LINKED ISSUE #65: UI deployment
After #19 #18

# Description
Add UI deployment with Terraform.
Might need to be refined after #18 and after #19

# Acceptance criteria
- [ ] UI can be deployed with other MVD components
- [ ] ~Steps to deploy and access the UI are documented~ (will be done in #19, since currently the build destroys the environment right away)
- [ ] UI is deployed in CD to test the terraform
- [ ] The UI can be used and can call the backend

# Tasks
- [ ] ~Build and push image to ACR in build~ (will be done in https://github.com/agera-edc/EDCDataDashboard/issues/1)
- [ ] Add UI deployment to Terraform
- [ ] ~Document steps to deploy the UI~
- [ ] Set up CORS
- [ ] Set backend URL

### LINKED ISSUE #94: UI deployment

### PR DESCRIPTION
**Pull request by algattik**
Deploy data dashboard webapp.

Closes agera-edc/MinimumViableDataspaceFork#94 

- Added CORS setup for browsers not to block cross-site calls.
- Added API key to authenticate requests from web app to backend API. This is also required for CORS setup in EDC..
- Added "inbox" storage account used by web app as destination for file transfers.

To verify what the web app is correctly deployed, you can use this deployment (where the Terraform destroy job was skipped): https://github.com/agera-edc/MinimumViableDataspace/actions/runs/2253208504 The URLs to the two web apps appear in the build annotations:
image
You can perform a data transfer. You can view the output blob using Azure portal (storage account 324company1inbox or 324company2inbox)

### CODE REVIEW

**Review by marcgs — APPROVED** [review]

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -264,5 +274,5 @@ jobs:
           TF_VAR_application_sp_client_id: dummy
           TF_VAR_application_sp_client_secret: dummy
           TF_VAR_registry_resource_group: ${{ secrets.COMMON_RESOURCE_GROUP }}
-          TF_VAR_registry_storage_account: ${{ secrets.REGISTRY_STORAGE_ACCOUNT }}
-          TF_VAR_registry_share: ${{ secrets.REGISTRY_SHARE }}
+          TF_VAR_registry_storage_account: ${{ secrets.CATALOG_STORAGE_ACCOUNT }}
+          TF_VAR_registry_share: ${{ secrets.CATALOG_SHARE }}
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml]: why value is for Catalog when variable named as registry?

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -95,11 +97,17 @@ jobs:
           KEY_VAULT=$(terraform output -raw key_vault)
           EDC_ACI_NAME=$(terraform output -raw edc_aci_name)
           RESOURCE_GROUP=$(terraform output -raw resource_group)
+          WEBAPP_URL=$(terraform output -raw webapp_url)
+          API_KEY=$(terraform output -raw api_key)
+          #TODO echo "::add-mask::${API_KEY}"
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml]: TODO still remains? or we want to address it now?

**Review by cpeeyush — COMMENTED** [review]
Comments raised.

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (algattik) [reply .github/workflows/cd.yaml]: Added add-mask

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (algattik) [reply .github/workflows/cd.yaml]: Fixed

**Review by cpeeyush — APPROVED** [review]

File: docs/developer/continuous_deployment.md [hunk 3]
```diff
@@ -97,3 +97,9 @@ Configure the following GitHub secrets:
 ### Deploying CD resources
 
 Manually run the `Initialize CD` GitHub Actions workflow.
+
+### Deploying Data Dashboard
+
+Fork the **EDC Data Dashboard** web app and run its deploy action.
```
Comment (marcgs) [inline docs/developer/continuous_deployment.md:103]: Is forking really required? Why not just pushing the image to an ACR?

Comment (algattik) [reply docs/developer/continuous_deployment.md:103]: Created https://github.com/agera-edc/MinimumViableDataspace/issues/85 so we can align this between the two repos

**Pull request merged**